modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/util/JsonHelperImpl.java

```diff
@@ -45,8 +46,17 @@ public JsonHelperImpl(JSONFactory jsonFactory) {
 
 	@Override
 	public boolean equals(String jsonArrayString1, String jsonArrayString2) {
-		JSONArray jsonArray1 = _toJSONArray(jsonArrayString1);
-		JSONArray jsonArray2 = _toJSONArray(jsonArrayString2);
+		JSONArray jsonArray1 = new JSONArrayImpl();
```

Comment:
@alecsloan I woudl arther write it as:

if((isEmpty(jsonArrayString1) && !isEmpty(jsonArrayString2)) ||
	(isEmpty(jsonArrayString2) && !isEmpty(jsonArrayString1))) {
	return false;
}

JSONArray jsonArray1 = _toJSONArray(jsonArrayString1);
JSONArray jsonArray2 = _toJSONArray(jsonArrayString2);

return JSONUtil.equals(jsonArray1, jsonArray2);